src/app/collective-rewards/allocations/page.tsx

```diff
@@ -37,6 +40,16 @@ export default function Allocations() {
     router.back()
   }
 
+  useEffect(() => {
+    if (!isConnected) {
+      router.replace('/')
+    }
+  }, [isConnected, router])
+
+  if (!isConnected) {
```

Comment:
I guess that this will show the Spinner while the `useEffect` executes the redirection, right?